components/script/dom/htmlcanvaselement.rs

```diff
@@ -156,6 +158,18 @@ impl LayoutHTMLCanvasElementHelpers for LayoutDom<HTMLCanvasElement> {
                 .unwrap_or(LengthOrPercentageOrAuto::Auto)
         }
     }
+
+    #[allow(unsafe_code)]
+    fn get_canvas_id_for_layout(&self) -> CanvasId {
+        unsafe {
+            let canvas = &*self.unsafe_get();
+            if let &Some(CanvasContext::Context2d(ref context)) = canvas.context.borrow_for_layout() {
+                context.get_canvas_id()
```

Comment:
Sorry, one more change is required. There needs to be a `get_canvas_id_for_layout` method added to `LayoutCanvasRenderingContext2DHelpers` in `canvasrenderingcontext2d.rs`. This line should then read `context.to_layout().get_canvas_id()`.